### PR TITLE
[dbex/59698] 526 Submit - Migrate GET call to /paymentInformation to Lighthouse PPIU

### PR DESCRIPTION
**Pull request by tommasina-va**
## Summary

- *This PR adds an abstracted PPIU/DirectDeposit service that can direct calls to GET /paymentInformation to a new Lighthouse PPIU Provider.*
- *Note: eVSS returns PaymentAddress in its response but Lighthouse has not implemented that in its response. We did create a new DisabilityCompensation::ApiProvider::PaymentAddress model so that it would be in place in case that's added in the future.*

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#59698

## Testing done

- *Until now, this GET request used the eVSS API to return payment information. This PR implements use of the Lighthouse Direct Deposit API under a Flipper feature toggle.*
- *New tests check that the correct API is used based on the Flipper setting*
- *New tests also check that the API response class and behavior is identical between the two APIs*

## What areas of the site does it impact?
*This affects the 526 submission GET call to /paymentInformation*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by aurora-a-k-a-lightning — DISMISSED** [review]
yaaaay! all TODOs for #59698 _**DONE!**_
thank you, @tommasina-va 😄

**Review by sethdarragile6 — DISMISSED** [review]

**Review by sethdarragile6 — DISMISSED** [review]

**Comment by tpharrison** [comment]
@tommasina-va @aurora-a-k-a-lightning The direct deposit service to LH has already been built by the authenticated experience team.  That service should probably be used to get account info for 526 submissions. We should probably discuss this.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1841,6 +1841,7 @@ spec/support/vcr_cassettes/lighthouse/auth/client_credentials/token_200.yml @dep
 spec/support/vcr_cassettes/lighthouse/benefits_claims/documents @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 spec/support/vcr_cassettes/lighthouse/benefits_claims/submit526 @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/dbex-trex
 spec/support/vcr_cassettes/lighthouse/claims/200_response.yml @department-of-veterans-affairs/Disability-Experience @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/dbex-trex
+spec/support/vcr_cassettes/lighthouse/direct_deposit @department-of-veterans-affairs/Disability-Experience @department-of-veterans-affairs/dbex-trex @department-of-veterans-affairs/backend-review-group
```
Comment (tpharrison) [inline .github/CODEOWNERS:1844]: The `vfs-authenticated-experience-backend` team is primarily responsible for this.

File: spec/support/vcr_cassettes/lighthouse/direct_deposit/show/errors/404_response.yml
Comment (tpharrison) [inline spec/support/vcr_cassettes/lighthouse/direct_deposit/show/errors/404_response.yml:1]: Duplicate of `direct_deposit/show/404_icn_not_found.yml`

**Review by tpharrison — COMMENTED** [review]
Code looks fine.  I just have some comments around code-ownership and duplication of vcr cassettes.

**Review by tpharrison — DISMISSED** [review]
🚀

**Review by sethdarragile6 — DISMISSED** [review]

**Review by tpharrison — APPROVED** [review]

**Review by kpethtel — DISMISSED** [review]
Mobile changes look fine.

**Review by aherzberg — DISMISSED** [review]
LGTM

**Review by tpharrison — APPROVED** [review]
🚀

**Review by aherzberg — APPROVED** [review]
LGTM for mobile

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**